### PR TITLE
Fix is_permitted with require_all = TRUE

### PR DESCRIPTION
**Pull request by Dentxinho**
This function doesn't work as expected if require_all is set to TRUE. It returns false on first loop.
Also, this function doesn't need a loop at all, this change make it more simple.

### CODE REVIEW

File: application/libraries/account/Authorization.php [hunk 1]
```diff
@@ -57,60 +57,38 @@ function is_permitted($permission_keys, $require_all = FALSE)
 
     $this->CI->load->model('account/Acl_permission_model');
 
-    $account_permissions = $this->CI->Acl_permission_model->get_by_account_id($account_id);
+    $account_permissions = array();
+
+    $permissions = $this->CI->Acl_permission_model->get_by_account_id($account_id);
+    foreach ($permissions as $perm) {
+        $account_permissions[] = strtolower($perm->key);
```
Comment (AdwinTrave) [inline application/libraries/account/Authorization.php]: I don't like the `strtolower` here at all. You are doing the same thing later on in `array_map`, but it also doesn't account for bad practice of having case sensitive permission keys.

Comment (Dentxinho) [inline application/libraries/account/Authorization.php]: I've added strtolower to both ``$permission_keys`` and ``$account_permissions`` to keep the same behaviour of ``strtolower($perm->key) == strtolower($key)``

Comment (AdwinTrave) [inline application/libraries/account/Authorization.php]: I understand that. I'm just pointing out that some people might have case sensitive permission keys (no matter if that is a good practice or not).

Comment (Dentxinho) [inline application/libraries/account/Authorization.php]: Done.

**Comment by AdwinTrave** [comment]
Do you want to update this to the current caching code or should I do it?

**Comment by Dentxinho** [comment]
@AdwinTrave, it is done ;)

**Comment by AdwinTrave** [comment]
Thank you!

**Pull request merged**